### DISCUSSION TITLE
What about expressions?

### DISCUSSION THREAD

**Discussion by TjShredder** [plugdata-team/plugdata#1850] [Ideas]
I guess it would need a PlugData specific object for input and output of expressions for VST3 and CLAP. I have a need for this in Bitwig, as Bitwig seems to like to translate Midi into Expressions and Expressions are always per note. This could be also a Topic about Midi 2.0...

**Comment by dromer** [comment]
Do you mean "Midi Polyphonic Expression"?

This would likely need very custom objects, since Puredata does not support this.

**Reply by TjShredder** [reply]
Yes, that was what I meant. 
Maybe its better to wait until Midi 2.0 is more spread, as obviously it would need special objects...

**Reply by dromer** [reply]
For sure, most DAWs don't support it yet I think. Would also make sense to start requesting this for Puredata as plugdata aims to be compatible with it.

**Reply by ben-wes** [reply]
it's possible to support MPE instruments with the current set of Pd objects. `midi-help.pd` includes a subpatch with some comments on that. i played around with a Haken Continuum a while ago and it wasn't hard to correctly interpret and use its MPE data. obviously, this is not Midi 2.0 though - that would certainly need different or updated objects and i know very little about it. :)

**Reply by dromer** [reply]
@ben-wes derp, I always forget that newer help files have these additional layers to them.
Apparently with some list trickery MPE can be used. (I can't use list object in my patches so have maybe mentally blocked this section haha)

MIDI 2.0 is a whole different story indeed.
